packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -57,6 +51,43 @@ export function isQueryFieldsValid(fields: Record<string, QueryBarFormField>) {
   );
 }
 
+function wrapFilterIfNotWrapped(
+  filter: QueryBarState['fields']['filter']
+): QueryBarState['fields']['filter'] {
+  if (validateField('filter', filter.string) !== false || !filter) {
+    return filter;
+  }
+
+  // When the filter isn't valid we try to wrap it in {}.
+  const filterWithBrackets = validateField('filter', `{${filter.string}}`);
+  if (filterWithBrackets !== false) {
+    return {
+      value: filterWithBrackets,
+      string: queryParser.stringify(filterWithBrackets),
+      valid: true,
+    };
+  }
+
+  return filter;
+}
+
+export function isQueryValid(state: QueryBarState) {
+  return QUERY_PROPERTIES.every((prop) => {
+    // When the filter is invalid but can be automatically altered to be valid,
+    // we want to allow the user to run the query.
+    if (prop === 'filter' && state.fields[prop].valid === false) {
```

Comment:
Is it a deliberate choice that we are doing this only for filter and not for other fields? I would be confused if special syntax allowed me to run queries for one case, but not for other